### ISSUE TITLE
Not detecting reCAPTCHAs for site "https://givingassistant.org"

### ISSUE THREAD

**Issue by rinoldsimon** [berstend/puppeteer-extra#48]
**Version:**
```
"puppeteer": "^1.17.0",
"puppeteer-extra": "^2.1.3",
"puppeteer-extra-plugin-recaptcha": "^3.0.4"
```

After login submit is clicked, the reCAPTCHA popup appears. Then I call `await page.solveRecaptchas()` and nothing happens.

Debugged a little bit and in the reCAPTCHA page I tried calling `let { captchas, error } = await page.findRecaptchas()` it returns `null`

So the plugin fails to detect the reCAPTCHA in the page.

**here is what is done:**
```
    // go to giving assistant page
    await page.goto('https://givingassistant.org');
    // click signin
    await page.waitForSelector('a.account-menu__button.user-signin');
    await page.evaluate(() => document.querySelector('a.account-menu__button.user-signin').click());
    //enter credentials
    await page.waitForSelector('#signin-email');
    await page.type('#signin-email', ga_email);
    await page.waitForSelector('#signin-password');
    await page.type('#signin-password', ga_password);
    //sign in button click
    await page.waitForSelector('#myModal_signin > div > form > div.modal-footer > button');
    await page.evaluate(() =>  document.querySelector("#myModal_signin > div > form > div.modal-footer > button").click())

    await page.waitFor(2000);

    // check whether signed in to GA
    // looking for the element is present in DOM
    let signedin = await page.$('li.anim_signedin > button > span > span > img')

    if (await signedin) {
      console.log("Login in to GivingAssistant successful")
      return true;
    } else {
      console.log("GivingAssistant - Redirects to reCAPTCHA page")
      // recaptcha solver
      let { captchas, error } = await page.findRecaptchas() // returns null
      // await page.solveRecaptchas() 
      ....
      ....
    }
```

**Comment by vans163** [comment]
For 1 `script[src="https://www.google.com/recaptcha/api.js"]` should be `script[src^="https://www.google.com/recaptcha/api.js"]`, but the problem goes deeper.

**Comment by gajus** [comment]
@berstend Any chance of adding what @vans163 shared to the main the package?

**Comment by eibrahim** [comment]
I agree.  This works well and I am using that fork but I prefer if we can have that merged into the main package instead of the fork.  thanks.

**Comment by berstend** [comment]
Happy to make changes to the current recaptcha plugin.

Unfortunately [the fork](https://github.com/MikhailGlv/puppeteer-extra-plugin-recaptcha-custom) made changes during their initial commit, so I need to reverse engineer what problem they're actually trying to solve and how.

**Comment by berstend** [comment]
> For 1 `script[src="https://www.google.com/recaptcha/api.js"]` should be `script[src^="https://www.google.com/recaptcha/api.js"]`

What is this supposed to do differently?

**Comment by berstend** [comment]
Closing due to inactivity.

**Comment by vans163** [comment]
> > For 1 `script[src="https://www.google.com/recaptcha/api.js"]` should be `script[src^="https://www.google.com/recaptcha/api.js"]`
> 
> @vans163 What is this supposed to do differently?

The ^ means to match the start of the string, without it it tries to match exactly, but sometimes there are query arguments following the api.js.  
```
https://www.google.com/recaptcha/api.js?query1=hi&two=2
```

**Comment by berstend** [comment]
That makes sense, thanks :) Will add it in a future update

**Comment by berstend** [comment]
Fixed in `puppeteer-extra-plugin-recaptcha@3.1.9` :)

**Comment by rinoldsimon** [comment]
I think this has not been fixed. Still, the newer version `puppeteer-extra-plugin-recaptcha@3.1.10` does not detect Recaptcha in `https://givingassistant.org` (as I mentioned in Question)
